src/site/content/en/blog/2022-state-of-css/index.md

```diff
@@ -0,0 +1,1902 @@
+
+### Was in `min-width` or `max-width`
+
+{% BrowserCompat 'css.at-rules.media.ranges' %}
+
+Before media query ranges, a CSS media query uses `min-width` and `max-width` to
+articulate over and under conditions. It may look like this:
+
+```css
+@media (min-width: 320px) {
+  …
+}
+```
+
+After media query ranges, the same media query could look like this:
+
+```css
+@media (320px >= width) {
+  …
+}
+```
+
+A CSS media query using both `min-width` and `max-width` may look like this:
+
+```css
+@media (min-width: 320px) and (max-width: 1280px) {
+  …
+}
+```
+
+After media query ranges, the same media query could look like this:
+
+```css
+@media (320px <= width <= 1280px) {
+  …
+}
+```
+
+Depending on your coding background, one of those will look much more legible
+than the other. Thanks to the spec additions, developers will be able to choose
+which they prefer, or even use them interchangeably.
+
+**Resources**  
+Spec: [https://www.w3.org/TR/mediaqueries-5/#mq-range-context](https://www.w3.org/TR/mediaqueries-5/#mq-range-context)  
+MDN: [https://developer.mozilla.org/docs/Web/CSS/Media_Queries/Using_media_queries#syntax_improvements_in_level_4](https://developer.mozilla.org/docs/Web/CSS/Media_Queries/Using_media_queries#syntax_improvements_in_level_4)  
+PostCSS plugin: [https://github.com/postcss/postcss-media-minmax](https://github.com/postcss/postcss-media-minmax)
+
+### No media query variables
+
+{% BrowserCompat 'css.at-rules.media.custom' %}
+
+Before `@custom-media`, media queries had to repeat themselves over and over, or
+rely on preprocessors to generate the proper output based on static variables
+during build time.
+
+After `@custom-media`, CSS allows aliasing media queries and the referencing of
+them, just like a custom property.
+
+Naming things is very important: it can align purpose with the syntax, making
+things easier to share and easier to use in teams. Here are a few custom media
+queries that follow me between projects:
+
+```css
+@custom-media --OSdark  (prefers-color-scheme: dark);
+@custom-media --OSlight (prefers-color-scheme: light);
+
+@custom-media --pointer (hover) and (pointer: coarse);
+@custom-media --mouse   (hover) and (pointer: fine);
+
+@custom-media --xxs-and-above (width >= 240px);
+@custom-media --xxs-and-below (width <= 240px);
+```
+
+Now that they're defined, I can use one of them like this:
+
+```css
+@media (--OSdark) {
+  :root {
+    …
+  }
+}
+```
+
+Find a [full list of custom media
+queries](https://open-props.style/#media-queries) I use inside my CSS custom
+property library [Open Props](https://open-props.style).
+
+**Resources**  
+Spec: [https://www.w3.org/TR/mediaqueries-5/#custom-mq](https://www.w3.org/TR/mediaqueries-5/#custom-mq)  
+PostCSS plugin: [https://github.com/postcss/postcss-custom-media](https://github.com/postcss/postcss-custom-media)  
+
+### Nesting selectors is so nice
```

Comment:
Can we mention PostCSS here? You have it in the links but it would be good as an aside.

Reply:
yep! done 👍🏻